src/Core/Types/Types/Descriptors/ObjectTypeDescriptor.cs

```diff
@@ -235,7 +235,9 @@ public IObjectTypeDescriptor IsOfType(IsOfType isOfType)
         public IObjectFieldDescriptor Field(NameString name)
         {
             ObjectFieldDescriptor fieldDescriptor =
-                Fields.FirstOrDefault(t => t.Definition.Name.Equals(name));
+                Fields.FirstOrDefault(t =>
+                    t.Definition.Name.Equals(name)
+                    && !t.Definition.Ignore);
```

Comment:
Fields is also used in the generic version because of `Field(x => x.Property)`

Reply:
But in that case it does not matter since the collision is only on names. I however think we should change the data structure to better controls ignored members.